zebra-scan/src/storage.rs

```diff
@@ -123,6 +141,11 @@ impl Storage {
             batch.insert_sapling_result(self, entry);
         }
 
+        // Add tracking entry for key.
+        if is_control_time {
+            batch.insert_sapling_height(self, &sapling_key, height);
```

Comment:
We can delete lower heights controls that were added before. That way we will only have at most 1 control result at a time in the database for each key.

Reply:
I am also unsure if the control block will show as a transaction found when getting results (via rocksdb tool or @upbqdn script).

Reply:
> We can delete lower heights controls that were added before. That way we will only have at most 1 control result at a time in the database for each key.

I think this might be out of scope for the MVP, it's in the Future Iterations list in #7728 as:
> Delete scanner results with empty transaction ID lists, except for the highest one for each key

Deleting results needs extra testing, so it might be easier to put them all in a later iteration.

Reply:
> I am also unsure if the control block will show as a transaction found when getting results (via rocksdb tool or @upbqdn script).

The control blocks will show as an empty transaction ID, one that is zero bytes long. So it's almost impossible to interpret them as a real transaction ID.

For the RocksDB tool we will need to specifically ignore them using a command line tool like grep.

Marek's tool calls a method which already ignores them here:
https://github.com/ZcashFoundation/zebra/blob/5dd33d7265b9a8afe7f6b84264cc4faa2a5bfada/zebra-scan/src/storage/db/sapling.rs#L89-L95

Reply:
After PR #8075 this is fine and incorrect code won't compile any more.